pkg/api/v1beta2/dynakube/dynakube_conversion.go

```diff
@@ -0,0 +1,113 @@
+package dynakube
+
+import (
+	"github.com/Dynatrace/dynatrace-operator/pkg/api/v1beta1/dynakube"
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/conversion"
+)
+
+// ConvertTo converts v1beta2 to v1beta1.
+func (src *DynaKube) ConvertTo(dstRaw conversion.Hub) error {
+	dst := dstRaw.(*dynakube.DynaKube)
+
+	dst.ObjectMeta = src.ObjectMeta
+
+	// DynakubeSpec
+	dst.Spec.APIURL = src.Spec.APIURL
+	dst.Spec.Tokens = src.Spec.Tokens
+	dst.Spec.CustomPullSecret = src.Spec.CustomPullSecret
+	dst.Spec.SkipCertCheck = src.Spec.SkipCertCheck
+	dst.Spec.Proxy = (*dynakube.DynaKubeProxy)(src.Spec.Proxy)
+	dst.Spec.TrustedCAs = src.Spec.TrustedCAs
+	dst.Spec.NetworkZone = src.Spec.NetworkZone
+	dst.Spec.EnableIstio = src.Spec.EnableIstio
+
+	// Status
+	dst.Status.Conditions = src.Status.Conditions
+
+	dst.Status.OneAgent.Instances = map[string]dynakube.OneAgentInstance{}
+
+	for key, value := range src.Status.OneAgent.Instances {
+		tmp := dynakube.OneAgentInstance{
+			PodName:   value.PodName,
+			IPAddress: value.IPAddress,
+		}
+		dst.Status.OneAgent.Instances[key] = tmp
+	}
+
+	dst.Status.OneAgent.Version = src.Status.OneAgent.Version
+
+	dst.Status.Phase = src.Status.Phase
+	dst.Status.UpdatedTimestamp = src.Status.UpdatedTimestamp
+
```

Comment:
related to my previous comment, we should consider using the new specific fields to add them as feature-flags here, so no information is lost during conversion.

Reply:
Then I think I did it correctly in https://github.com/Dynatrace/dynatrace-operator/pull/3092/commits/d3318cdcd1da4630154e41f06ca2c18ba5e6e48d

Reply:
you are always creating a new map for the Annotations so only the last will take effect.
should just add the key-values, like`dst.Annotations[AnnotationFeatureOneAgentSecCompProfile] = hostMonitoring.SecCompProfile`

also be sure to take care of the `dst.Annotations == nil` scenario 😉